lib/daru/index.rb

```diff
@@ -50,15 +50,17 @@ def map(&block)
     attr_reader :relation_hash, :size
 
     def initialize index
-      index = 0                         if index.nil?
-      index = Array.new(index) { |i| i} if index.is_a? Integer
-      index = index.to_a                if index.is_a? Daru::Index
-
-      @relation_hash = {}
-      index.each_with_index do |n, idx|
-        @relation_hash[n] = idx 
+      index = case index
+        when nil
+          []
+        when Integer
+          index.times.to_a
+        when Enumerable
+          index.to_a
```

Comment:
Sorry for being pedantic, but I think we still need `else` here, or we have pretty cryptic error:
```ruby
Daru::Index.new('foo')
# => NoMethodError: undefined method `each_with_index' for nil:NilClass
```